### PR TITLE
Update to Rust1.80.0

### PR DESCRIPTION
**Pull request by Hmikihiro**
rustc library had some changes in 1.79 & 1.80.

`LazyLock` is part of standard library in 1.80
Rust1.80 can replace from `lazy_static` & `once_cell::sync::Lazy` to `std::LazyLock`.
I would suggest updating to rust1.80 as the first step, with replace to `LazyLock` later.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] There are tests for these changes (test-unit crown)

### CODE REVIEW

File: support/crown/src/common.rs [hunk 1]
```diff
@@ -49,22 +48,22 @@ pub fn in_derive_expn(span: Span) -> bool {
...
+
+         impl Symbols {
+             fn new() -> Self {
+                 Symbols {
+                     $( $s: Symbol::intern(stringify!($s)), )+
+                 }
+             }
+         }
+     }
+ }
```
Comment (sagudev) [inline support/crown/src/common.rs:66]: This is accident I assume (normal formatting has 4 spaces not 5).

File: support/crown/src/trace_in_no_trace.rs [hunk 2]
```diff
@@ -30,7 +30,7 @@ declare_tool_lint! {
 
 const EMPTY_TRACE_IN_NO_TRACE_MSG: &str =
     "must_not_have_traceable marked wrapper is not needed for types that implements \
-empty Traceable (like primitive types). Consider removing the wrapper.";
+  empty Traceable (like primitive types). Consider removing the wrapper.";
```
Comment (sagudev) [inline support/crown/src/trace_in_no_trace.rs:33]: why?

**Review by sagudev — COMMENTED** [review]
LGTM, commits have great messages, but there are some formatting changes I am not sure of (although tidy is happy?)

File: support/crown/src/common.rs (hunk 1 above)
Comment (Hmikihiro) [reply support/crown/src/common.rs:66]: Sorry, There are not intended. Thank you for your advice.

**Pull request closed without merging**